contrib/pkg/labelobjects/labelobjects.go

```diff
@@ -0,0 +1,928 @@
+		"installLogPVCsFailureCount":                  lm.installLogPVCsFailureCount,
+		"imagesetJobsFailureCount":                    lm.imagesetJobsFailureCount,
+		"clusterDeprovisionsFailureCount":             lm.clusterDeprovisionsFailureCount,
+		"dnsZonesFailureCount":                        lm.dnsZonesFailureCount,
+		"mergedPullSecretsFailureCount":               lm.mergedPullSecretsFailureCount,
+		"clusterDeprovisionJobsFailureCount":          lm.clusterDeprovisionJobsFailureCount,
+		"clusterProvisionJobsFailureCount":            lm.clusterProvisionJobsFailureCount,
+		"clusterStatesFailureCount":                   lm.clusterStatesFailureCount,
+		"controlPlaneCertificateSyncSetsFailureCount": lm.controlPlaneCertificateSyncSetsFailureCount,
+		"remoteIngressSyncSetsFailureCount":           lm.remoteIngressSyncSetsFailureCount,
+		"identityProviderSyncSetsFailureCount":        lm.identityProviderSyncSetsFailureCount,
+		"kubeconfigSecretsFailureCount":               lm.kubeconfigSecretsFailureCount,
+		"kubeAdminCredsSecretsFailureCount":           lm.kubeAdminCredsSecretsFailureCount,
+		"syncSetInstancesFailureCount":                lm.syncSetInstancesFailureCount,
+		"selectorSyncSetInstancesFailureCount":        lm.selectorSyncSetInstancesFailureCount,
+	}).Info("Failed Object Labelings")
+
+	if !succeeded {
+		return errors.New("Label migration failed")
+	}
+
+	return nil
+}
+
+func (l *labelMigrator) labelClusterProvisions() error {
+	all := &hivev1alpha1.ClusterProvisionList{}
+
+	err := l.client.List(context.TODO(), all)
+	if err != nil {
+		l.clusterProvisionsFailureCount++
+		return errors.Wrap(err, "Failed listing ClusterProvisions")
+	}
+
+	for _, item := range all.Items {
+		objLogger := l.newObjectLogger(item.Name, item.Namespace, item.Kind)
+
+		name, found := findControllerOwnerNameByKind(&item, "ClusterDeployment")
+		if !found {
+			objLogger.Error("Missing ClusterDeployment owner reference")
+			continue
+		}
+
+		item.Labels = k8slabels.AddLabel(item.Labels, constants.ClusterDeploymentNameLabel, name)
+
+		if l.updateObject(objLogger.WithField("labels", item.Labels), &item) {
+			l.clusterProvisionsSuccessCount++
+		} else {
+			l.clusterProvisionsFailureCount++
+		}
+	}
+
+	if l.clusterProvisionsFailureCount > 0 {
+		return fmt.Errorf("Failed updating %v object(s)", l.clusterProvisionsFailureCount)
+	}
+
+	return nil
+}
+
+func (l *labelMigrator) labelInstallLogPVCs() error {
+	all := &corev1.PersistentVolumeClaimList{}
+
+	err := l.client.List(context.TODO(), all)
+	if err != nil {
+		l.installLogPVCsFailureCount++
+		return errors.Wrap(err, "Failed listing PersistentVolumeClaims")
+	}
+
+	for _, item := range all.Items {
+		objLogger := l.newObjectLogger(item.Name, item.Namespace, "PersistentVolumeClaim")
+
+		val, ok := item.Labels[constants.InstallJobLabel]
+		if !ok || val != "true" {
+			// Not a PVC that we're interested in. Move on.
+			continue
+		}
```

Comment:
I think you could bypass this label check if you wanted by filtering in the List: https://github.com/openshift/hive/blob/master/pkg/controller/metrics/metrics.go#L232